app/scripts/metamask-controller.js

```diff
@@ -987,11 +987,16 @@ export default class MetamaskController extends EventEmitter {
 
     const getSnapController = () => this.snapController;
 
+    const persistAndUpdateAccounts = async () => {
+      await this.keyringController.persistAllKeyrings();
+      await this.accountsController.updateAccounts();
```

Comment:
Can you please add a comment explaining why we need to call `updateAccounts` too?

Reply:
Comment added here [1b4f00a](https://github.com/MetaMask/metamask-extension/pull/23178/commits/1b4f00a9144f4db6a56b21894a26a0d498b89bbc)